shared/src/commonMain/kotlin/org/hyperskill/app/sentry/domain/SentryExtentions.kt

```diff
@@ -22,6 +23,8 @@ suspend inline fun <T> SentryInteractor.withTransaction(
         val result = measureBlock()
         finishTransaction(transaction)
         result
+    } catch (e: CancellationException) {
+        throw e
```

Comment:
@XanderZhu 

1. If `CancellationException` catched and then rethrows, could this lead to a runtime crash?
2. I suppose the Sentry transaction will never be finished

Reply:
@ivan-magda 
> If CancellationException is caught and then rethrows, could this lead to a runtime crash?

No, it couldn't. `CancellationException` is a special exception that is used in coroutines to cancel running coroutines and it shouldn't be watched.
We use `CoroutineExceptionHandler` to handle all the exceptions inside the ActionDisptacher coroutines. (see [code](https://github.com/hyperskill/mobile-app/blob/6c05f421f1b6e768954f82ce77d0dcdf926b7613/shared/src/commonMain/kotlin/org/hyperskill/app/core/presentation/ActionDispatcherOptions.kt#L18-L22)). `CoroutineExceptionHandler` handles all the exceptions except of `CancellationException`.
https://kotlinlang.org/docs/exception-handling.html#supervision-job
https://stackoverflow.com/a/76261124

Reply:
@ivan-magda 
> I suppose the Sentry transaction will never be finished

Good point. We should finish sentry transition for `CancellationException` too.